### PR TITLE
Merge refactors

### PR DESCRIPTION
**Pull request by webthethird**
@omurovec the main difference here is that we try to read/write the collaterals from/to the cache.

### CODE REVIEW

File: src/factory/providers/zksync-era/increment/index.ts [hunk 1]
```diff
@@ -23,27 +40,39 @@ async function tvl(params: ITvlParams): Promise<Partial<ITvlReturn>> {
...
+      collaterals,
+      'cache/collaterals.json',
+      chain,
+      provider,
+    );
+  }
 
   const balanceResults = await util.getTokenBalances(
     VAULT_ADDRESS,
-    collaterals.map((collateral) => collateral.asset),
```
Comment (webthethird) [inline src/factory/providers/zksync-era/increment/index.ts:38]: I was getting the following error before changing `collateral.asset` to `collateral[0]`:
`{"statusCode":500,"message":"Cannot read properties of undefined (reading 'balance')"}`

File: src/factory/providers/zksync-era/increment/index.ts [hunk 2]
```diff
@@ -1,18 +1,35 @@
...
 const START_BLOCK = 28541484;
 const VAULT_ADDRESS = '0x703500cAF3c79aF68BB3dc85A6846d1C7999C672';
 const USDC = '0x3355df6D4c9C3035724Fd0e3914dE96A5a83aaf4';
+type Collateral = {
+  asset: string;
+  weight: BigInt;
+  decimals: number;
+  currentAmount: BigInt;
+  maxAmount: BigInt;
+};
```
Comment (webthethird) [inline src/factory/providers/zksync-era/increment/index.ts]: Given the comment above, I'm not sure this type definition does any good here

File: src/factory/providers/zksync-era/increment/index.ts (hunk 1 above)
Comment (omurovec) [reply src/factory/providers/zksync-era/increment/index.ts:38]: Ah ok, yeah that makes sense. The Collateral struct was just not decoded properly, your solution works best

File: src/factory/providers/zksync-era/increment/index.ts (hunk 2 above)
Comment (omurovec) [reply src/factory/providers/zksync-era/increment/index.ts]: Yeah, this would not be needed.

**Review by omurovec — DISMISSED** [review]
Let's just remove the collaterals definition and we should be good to go

**Pull request merged**